pkg/reconciler/trigger/trigger.go

```diff
@@ -162,13 +172,21 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, t *v1beta1.Trigger) pkgr
 		},
 	)
 
-	if err := r.reconcileDispatcherDeployment(ctx, t, subscriberURI); err != nil {
+	deployment, err := r.reconcileDispatcherDeployment(ctx, t, subscriberURI)
+	if err != nil {
 		logging.FromContext(ctx).Error("Problem reconciling dispatcher Deployment", zap.Error(err))
 		t.Status.MarkDependencyFailed("DeploymentFailure", "%v", err)
 		return err
 	}
 
-	return nil
+	_, err = r.reconcileScaleTriggerSecret(ctx, rabbitmqURL, t)
```

Comment:
ok, it took me almost 30 minutes to understand why that part was needed but after digging in KEDA codebase, I finally get it 😅

Reply:
We should be able to use TriggerAuthentication rather than needing to add this additional secret to the target deployment of the scaler, but I don't believe that's implemented yet in the RabbitMq scaler. I didn't try, but in the code it seemed to always look for the value of the `host` param from the ScaledObject "triggers"  map and then check that against the Secrets it discovered on the target. Does that align with what you noticed?

Reply:
I think only the trigger Metadata is checked against the secrets.
The AuthenticationRef `host` param is indeed looked up but the value is directly used to populate the hostname later used to connected to rabbit.

Reply:
I need to dig more into `scale_handler#parseAuthRef`, there are so many levels of indirections it's hard to get the whole picture...

Reply:
If the trigger references a TriggerAuthentication with a secret ref containing the host, I think it _would_ work.